draft-mohan-asdf-sdf-protocol-mapping.md

```diff
@@ -323,9 +337,118 @@ For example, a Zigbee protocol mapping for a temperature property might look lik
 
 ## IP based Protocol Mapping
 
-The protocol mapping mechanism can potentially also be used for IP-based protocols
-such as HTTP or CoAP. An example of a protocol mapping for a property using HTTP
-might look like:
+The protocol mapping mechanism can also be used for IP-based protocols
+such as HTTP or CoAP.
+
+### HTTP
+
+The HTTP protocol mapping allows SDF models to specify how properties, actions,
+and events should be accessed using the HTTP protocol. The mapping includes
+details such as method, path, optional headers, request body and query
+parameters that are used to access the corresponding SDF affordances.
+
+~~~ cddl
+{::include cddl/http-protocol-map.cddl}
+~~~
+{: #httpmap1 title="CDDL definition for HTTP Protocol Mapping for properties, actions, and events"}
+
+Where:
+
+- `method` is the method of the HTTP request towards the affordance. Allowed methods are `GET`, `POST`, `PUT`, and `DELETE`.
+- `path` is the path targeted by the HTTP request.
+- `query` is the optional query string expressed as a text map. The key of the map is the name of the query parameter and the value of the map is the value of the query parameter.
+- `headers` is the optional set of HTTP headers expressed as text map. The key of the map is the name of the header and the value of the map is the value of the header field.
+- `body` is the optional body of the HTTP request expressed as text.
+
+For example, a HTTP protocol mapping for a temperature property might look like:
+
+~~~ jsonc
+{
+  "sdfProperty": {
+    "temperature": {
+      "sdfProtocolMap": {
+        "http": {
+          "method": "GET",
+          "path": "/device/123/temperature/0",
+          "query": {"unit": "celsius"},
+        }
+      }
+    }
+  }
+}
+~~~
+
+The above SDF Protocol Mapping will be resolved to the following HTTP request: `GET /device/123/temperature/0?unit=celsius`.
+
+### CoAP
+
+The CoAP protocol mapping allows SDF models to specify how affordances should be
+accessed using the CoAP protocol. The mapping includes details such as method,
+URI, as well as optional query string and options.
+
+~~~ cddl
+{::include cddl/coap-protocol-map.cddl}
+~~~
+{: #coapmap1 title="CDDL definition for CoAP Protocol Mapping for properties, actions, and events"}
+
+Where:
+
+- `method` is the method of the CoAP request towards the affordance. Allowed methods are `GET`, `POST`, `PUT`, and `DELETE`.
```

Comment:
For CoAP, there is also `PATCH`, `FETCH`, and `iPATCH` defined. Should these be disallowed to be used here? Or would it make sense to refer to the respective IANA registry here?

I guess similar considerations apply to HTTP.

Reply:
Maybe the current list of entries could be given as an example, while mentioning that the names assigned to the method codes in the [registry](https://www.iana.org/assignments/core-parameters/core-parameters.xhtml#method-codes) are authoritative?